### PR TITLE
Add Warning To fix_formatting.sh Script When Using -a Argument

### LINKED ISSUE #791: Add Warning To fix_formatting.sh Script When Using -a Argument
### Description of the task
The `-a` flag on the `fix_formatting.sh` script will format the entire codebase and can be very slow. A more intelligent way to use the script is like `fix_formatting.sh -b master` which will format only the files that are different between the current branch and `master` (or any other specified branch). We should add a warning to the `-a` flag, so that interactions look like
```
./clang_format/fix_formatting.sh -a
WARNING: "-a" will format all the files and can be very slow! 
Please consider using "-b" instead!
Are you sure you want to format all files (y/n)?
```

### Acceptance criteria

- [ ] Add a warning and confirmation to the `-a` flag of the `fix_formatting.sh` script as described above.

### PR DESCRIPTION
**Pull request by amrm3lm**
### Description
Added a warning to the -a flag in the `fix_formatting.sh` script. Following the warning, the script prompts the user on whether they want to continue execution or not. If the response is 'n', script exits. 

### Testing Done
Tested in terminal for cases where the user inputs 'y', 'n', or something invalid. Also tested the other commands just in case. All pass.

### Resolved Issues
resolves #791 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: clang_format/fix_formatting.sh [hunk 1]
```diff
@@ -54,24 +54,35 @@ while test $# -gt 0; do
...
-
-            shift
-            ;;
+            echo "WARNING: "-a" will format all the files and can be very slow! Please consider using "-b" instead!"
+
+            echo "Are you sure you want to format all files (y/n)?"
+            read continueFormat
+            if [ "$continueFormat" == "n" ]; then
+		exit 0
+	    elif [ "$continueFormat" == "y" ]; then
```
Comment (jonathanlew) [inline clang_format/fix_formatting.sh]: Indenting

**Review by jonathanlew — CHANGES_REQUESTED** [review]

File: clang_format/fix_formatting.sh (hunk 1 above)
Comment (MathewMacDougall) [reply clang_format/fix_formatting.sh]: Specifically the indenting is inconsistent, and there seem to be tabs mixed in as well. Please indent using 4 spaces.

**Review by garethellis0 — COMMENTED** [review]
Nothing to add here.

**Comment by MathewMacDougall** [comment]
@amrm3lm any updates on this?

File: clang_format/fix_formatting.sh (hunk 1 above)
Comment (amrm3lm) [reply clang_format/fix_formatting.sh]: I fixed indenting in the most recent push

**Review by akhilveeraghanta — APPROVED** [review]
LGTM

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by jonathanlew — APPROVED** [review]
Good work!

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**